src/formatters/utils/sortResults.ts

```diff
@@ -1,13 +1,54 @@
 import { IRuleResult } from '../../types';
 
-export const sortResults = (results: IRuleResult[]) => {
-  return [...results].sort((resultA, resultB) => {
-    const diff = resultA.range.start.line - resultB.range.start.line;
+const safeCastToString = (code: string | number | undefined): string => {
+  if (code === undefined) {
+    throw new Error('Unexpected undefined code');
+  }
+
+  if (typeof code === 'number') {
+    throw new Error('Unexpected numbered type code');
+  }
+
+  return code;
+};
+
+const ARTIFICIAL_ROOT = 'root';
+
+const safeSource = (result: IRuleResult): string => {
+  const source = result.source === void 0 ? ARTIFICIAL_ROOT : result.source;
+  return source;
+};
+
+export const resultsComparer = (resultA: IRuleResult, resultB: IRuleResult): number => {
```

Comment:
Let's make it return `-1`, `0` or `1`
```suggestion
export const resultsComparer = (resultA: IRuleResult, resultB: IRuleResult): number => {
```

Reply:
@P0lip Fixed.